### DISCUSSION TITLE
Clarity on common Storage use case: URLs for public images

### DISCUSSION THREAD

**Discussion by seenickcode** [supabase/supabase#1462] [Questions] [answered]
Hello, a common use case for web or mobile app is to access public URLs for images. 

1. After reading the Storage documentation, it is not clear to me if the `token?=<token>` in the URL for the public Storage object is something that is static or if it can change over time? Is it indeed static and I can just hard code this URL path + query string into my web or mobile app?

2. It is not clear if I _must_ create a Object policy or not. It seems that I don't have to because Storage offers a public URL already. Is that right?

3. Just curious, does Storage leverage a CDN?

Thanks,

**Comment by inian** [comment, marked as answer]
Hi @seenickcode ,

1. A different `token` is returned everytime you call the `createSignedUrl` function. If you generate a signed URL with an expiry of 1 month, it will continue to be valid for 1 month from the point you generated the signed URL. You can set a long expiry and put this in your website or mobile app. 
2. If the client directly is using the signedURL, you don't need to create a policy for that client. If your server is generating the signed URLs and the server is using the `service key` to do that, no policies are required at all since the service key bypasses row level security. 
3. Not yet, but we are working on this. Tentative launch in ~3 months.

**Reply by inian** [reply]
For anyone coming across this, we have [launched public buckets](https://supabase.io/docs/guides/storage#accessing-objects) now. You should be able to access assets in a public bucket without a token or a Authorization header.

**Reply by SaraVieira** [reply]
Amazing!!!! 🎉🎉🎉

**Comment by seenickcode** [comment]
Thank you, Sean!

On May 6, 2021, Sean Pianka ***@***.***> wrote:
> Hi @seenickcode <https://github.com/seenickcode> ,
>
> 1. A different token is returned everytime you call the
>    createSignedUrl function. If you generate a signed URL with an
>    expiry of 1 month, it will continue to be valid for 1 month from
>    the point you generated the signed URL. You can set a long expiry
>    and put this in your website or mobile app.
> 2. If the client directly is using the signedURL, you don't need to
>    create a policy for that client. If your server is generating the
>    signed URLs and the server is using the service key to do that, no
>    policies are required at all since the service key bypasses row
>    level security.
> 3. Not yet, but we are working on this. Tentative launch in ~3 months.
>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/supabase/supabase/discussions/1462#discussioncomment-
707401>, or unsubscribe <https://github.com/notifications/unsubscribe-
auth/AAASQOKR4ZNAQU2R45W62O3TMNWJNANCNFSM44IFXX5A>.

**Comment by SaraVieira** [comment]
Sorry to barge in, just want to ask, this means there is no way of having a bucket where all the files are just publicly accessible like in aws?

I would always need a token right? 

My use case is have people be able to copy a url to a jpg file and directly use it on their threejs application

**Reply by inian** [reply]
Hey @seenickcode, we will be launching this feature in the next 3 days or so.

**Reply by SaraVieira** [reply]
Hey!

Was that answer for me? If so I'll then keep adding my stuff and wait for the feature

**Reply by inian** [reply]
oops, tagged the main OP by mistake. Yup meant to tag you Sara! I will update here again once the feature is launched

**Reply by SaraVieira** [reply]
Awesome! Great news and thank you so much for the quick answer 🙏

**Reply by inian** [reply]
Hey @SaraVieira, this feature has been [launched](https://supabase.io/docs/guides/storage#accessing-objects) now.
